doc/ipython-notebooks/multiclass/KNN.ipynb

```diff
@@ -260,14 +260,110 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# The data set is Ronald Fisher's measurements of type, petal width (PW), petal length (PL), sepal width (SW), \n",
+    "# and sepal length (SL) for a sample of 150 irises. \n",
+    "# Check more information in http://www.math.uah.edu/stat/data/Fisher.html\n",
+    "low_dimension_martix = np.loadtxt('../../../data/multiclass/Fisher.txt')\n",
+    "\n",
+    "Xall_low = low_dimension_martix[:,[1, 2, 3, 4]].transpose()\n",
+    "Yall_low = low_dimension_martix[:, 0]\n",
+    "\n",
+    "Xtrain_low = Xall_low[:, :100]\n",
+    "Ytrain_low = Yall_low[:100]\n",
+    "\n",
+    "Xtest_low = Xall_low[:, 100:]\n",
+    "Ytest_low = Yall_low[100:]\n",
+    "\n",
+    "print Xall_low.shape\n",
+    "print Xtrain_low.shape\n",
+    "print Xtest_low.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "labels_low = MulticlassLabels(Ytrain_low)\n",
+    "feats_low  = RealFeatures(Xtrain_low)\n",
+    "k=3\n",
+    "dist_low = EuclideanDistance()\n",
+    "knn_low = KNN(k, dist_low, labels_low)\n",
+    "labels_test_low = MulticlassLabels(Ytest_low)\n",
+    "feats_test_low  = RealFeatures(Xtest_low)\n",
+    "knn_low.train(feats_low)\n",
+    "pred_low = knn_low.apply_multiclass(feats_test_low)\n",
+    "print \"Predictions\", pred_low[:5]\n",
+    "print \"Ground Truth\", Ytest_low[:5]\n",
+    "\n",
+    "evaluator_low = MulticlassAccuracy()\n",
+    "accuracy_low = evaluator_low.evaluate(pred_low, labels_test_low)\n",
+    "\n",
+    "print \"Accuracy = %2.2f%%\" % (100*accuracy_low)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "start = Time.get_curtime()\n",
+    "knn_low.set_k(3)\n",
+    "knn_low.set_knn_solver_type(KNN_BRUTE)\n",
+    "pred_low = knn_low.apply_multiclass(feats_test_low)\n",
+    "print \"Standard KNN took %2.4fs\" % (Time.get_curtime() - start)\n",
+    "\n",
+    "\n",
+    "start = Time.get_curtime()\n",
+    "knn_low.set_k(3)\n",
+    "knn_low.set_knn_solver_type(KNN_COVER_TREE)\n",
+    "pred_low = knn_low.apply_multiclass(feats_test_low)\n",
+    "print \"Covertree KNN took %2.4fs\" % (Time.get_curtime() - start)\n",
+    "\n",
+    "\n",
+    "start = Time.get_curtime()\n",
+    "knn_low.set_k(3)\n",
+    "knn_low.set_knn_solver_type(KNN_KDTREE)\n",
+    "pred_low = knn_low.apply_multiclass(feats_test_low)\n",
+    "print \"KDTree KNN took %2.4fs\" % (Time.get_curtime() - start)"
```

Comment:
All the algrotihms are super fast, which makes timing very inaccurate. Can you maybe use a larger dataset where things take at least order of a second?